sdk/textanalytics/ai-text-analytics/README.md

```diff
@@ -0,0 +1,543 @@
+
+```typescript
+const documents = [
+  "I hated the movie. It was so slow!",
+  "The movie made it into my top ten favorites.",
+  "What a great movie!"
+];
+```
+
+or, if you wish to pass in a per-item document `id` or `language`/`countryHint`, they can be given as a list of `TextDocumentInput` or `DetectLanguageInput` depending on the operation;
+
+```javascript
+const textDocumentInputs = [
+  { id: "1", language: "en", text: "I hated the movie. It was so slow!" },
+  { id: "2", language: "en", text: "The movie made it into my top ten favorites." },
+  { id: "3", language: "en", text: "What a great movie!" }
+];
+```
+
+See [service limitations][data_limits] for the input, including document length limits, maximum batch size, and supported text encodings.
+
+### Return Value
+
+The return value corresponding to a single document is either a successful result or an error object. Each `TextAnalysisClient` method returns a heterogeneous array of results and errors that correspond to the inputs by index. A text input and its result will have the same index in the input and result collections. The collection may also optionally include information about the input batch and how it was processed in the `statistics` field.
+
+An **result**, such as `AnalyzeSentimentResult`, is the result of a Language operation, containing a prediction or predictions about a single text input. An operation's result type also may optionally include information about the input document and how it was processed.
+
+The **error** object, `TextAnalysisErrorResult`, indicates that the service encountered an error while processing the document and contains information about the error.
+
+### Document Error Handling
+
+In the collection returned by an operation, errors are distinguished from successful responses by the presence of the `error` property, which contains the inner `TextAnalysisError` object if an error was encountered. For successful result objects, this property is _always_ `undefined`.
+
+For example, to filter out all errors, you could use the following `filter`:
+
+```javascript
+const results = await client.analyze("SentimentAnalysis", documents);
+const onlySuccessful = results.filter((result) => result.error === undefined);
+```
+
+**Note**: TypeScript users can benefit from better type-checking of result and error objects if `compilerOptions.strictNullChecks` is set to `true` in the `tsconfig.json` configuration. For example:
+
+```typescript
+const [result] = await client.analyze("SentimentAnalysis", ["Hello world!"]);
+
+if (result.error !== undefined) {
+  // In this if block, TypeScript will be sure that the type of `result` is
+  // `TextAnalysisError` if compilerOptions.strictNullChecks is enabled in
+  // the tsconfig.json
+
+  console.log(result.error);
+}
+```
+
+This capability was introduced in TypeScript 3.2, so users of TypeScript 3.1 must cast result values to their corresponding success variant as follows:
```

Comment:
I wonder if we can drop this? 3.1 came out in September of 2018

Reply:
Good point, done!